c/src/ml-api-service-remote.c

```diff
@@ -283,17 +273,36 @@ _mlrs_model_register (gchar * service_key, nns_edge_data_h data_h,
     _ml_loge ("Failed to register model, service ket:%s", service_key);
     ret = FALSE;
   }
-error:
-  g_free (current_dir);
-  g_free (dir_path);
-  g_free (activate);
-  g_free (model_path);
-  g_free (description);
-  g_free (name);
 
   return ret;
 }
 
+/**
+ * @brief Get path to save the model given from remote sender.
+ * @note The caller is responsible for freeing the returned data using g_free().
+ */
+static gchar *
+_mlrs_get_model_dir_path (_ml_remote_service_s * remote_s,
+    const gchar * service_key)
+{
+  g_autofree gchar *dir_path = NULL;
+
+  if (remote_s->path) {
+    dir_path = g_strdup (remote_s->path);
+  } else {
+    g_autofree gchar *current_dir = g_get_current_dir ();
+
+    dir_path = g_build_path ("/", current_dir, service_key, NULL);
```

Comment:
recommend using G_DIR_SEPARATOR_S instead '/'  :)

Reply:
Thanks, I changed them and other files.